src/veh_interact.cpp

```diff
@@ -643,7 +643,11 @@ void veh_interact::do_repair()
             int xOffset = veh->parts[p].mount.y + ddy;
             int yOffset = -(veh->parts[p].mount.x + ddx);
 
-            move_cursor(xOffset, yOffset);
+            if (vertical_menu) {
+                move_cursor(xOffset, yOffset);
+            } else {
+                move_cursor(-yOffset, xOffset);
+            }
```

Comment:
Since you're here anyway, you may as well make `xOffset` actually calculate an offset on the vehicular X axis, with the same happening to `yOffset` on the Y axis, and re-sign and re-order as needed:

```C++
int xOffset = veh->parts[p].mount.x + ddx;
int yOffset = veh->parts[p].mount.y + ddy;

if (vertical_menu) {
    move_cursor(-yOffset, xOffset);
} else {
    move_cursor(xOffset, yOffset);
}
```

As I understand it, the vertical menu vehicle rotation is the one that's non-canonical, so we shouldn't be judging X and Y based on its orientation.